scanner/base.go

```diff
@@ -782,9 +782,6 @@ func AnalyzeLibrary(ctx context.Context, path string, contents []byte, filemode
 	if err != nil {
 		return nil, xerrors.Errorf("Failed to prepare filesystem for post-analysis. err: %w", err)
 	}
-	defer func() {
-		_ = composite.Cleanup()
-	}()
```

Comment:
As described in https://github.com/aquasecurity/trivy/pull/9241

> Normal operation: Existing defer os.Remove statements handle cleanup when functions complete

in normal cases, I guess, timely cleanup is preferable, for example, scanning many servers creates many temp files and should be cleaned up per server.

In the moment, composite.Cleanup() does NOT cleanup `analyzer-file-*`, I think it is a bug of composite module and should be fixed.